DataDog/datadog-agent
# Resolve Traps OIDs to names

With this PRs traps payload format has been refactored:
1. The source and service are now `snmp-traps`.
2. The traps variables are now stored under `.raw_variables` instead of `.variables`.
3. Traps payloads are enriched thanks to an OIDResolver interface that loads traps_db files located in `snmp.d/traps_db/`

# How to test:
1. If not shipped already, you can download this [database file](https://s3.amazonaws.com/dd-agent-omnibus/snmp_traps_db/dd_traps_db-0.1.0.json.gz) and place it under `snmp.d/traps_db/` directory (you can leave it compressed).
2. Start the agent with the following configuration in `datadog.yaml`

```yaml
logs_enabled: true
snmp_traps_enabled: true
snmp_traps_config:
  port: 9162
  community_strings:
    - 'public'
    - 'default'
```

3. Send traps on port 9162 with the command `snmptrap -v2c -c public localhost:9162 '' IF-MIB::linkDown ifIndex i 12 ifAdminStatus i 1 ifOperStatus i 2`
4. Look on Datadog logs and check that the received traps payload have been correctly formatted and enriched.
